### PR TITLE
Convert Register form from JS to TS

### PR DESCRIPTION
**Pull request by wariofdz**
### Changed:

- Added new enum on `RegisterStatus`.
- Adden new type on `FormFieldsRegister`.
- Added new types on `UseRegisterFormResult`.
- Added new type `RegisterResponse` for `httpClient.createUser`.
- Added new type `RegisterSerializedResponse` for serialized response to `RegisterResponse` type.
- Transformed `Register` component and his test from JSX to TSX.
- Transformed `useRegisterForm` hook and his test from JS to TS.
- Changed name of file `FormsFieldsApp` for `FormsFieldLogin`.
- Changed file `FormsFieldsApp` to `FormsFieldsLogin` on `useLoginForm` and `UseLoginFormResult`.

### CODE REVIEW

File: packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import { buildSerializableResponse } from '../../common/http/helpers/buildSerializableResponse';
+import { Either } from '../../common/models/Either';
+import { UseRegisterFormParams } from '../models/UseRegisterFormResult';
+import { RegisterStatus } from '../models/RegisterStatus';
+import { FormValidationResult } from '../models/FormValidationResult';
+import { FormFieldsRegister } from '../models/FormFieldsRegister';
+import { RegisterResponse } from '../../common/http/models/RegisterResponse';
+import { RegisterSerializedResponse } from '../../common/http/models/RegisterSerializedResponse';
+
+export const INVALID_CREDENTIALS_REG_ERROR: string = 'Invalid credentials.';
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts:16]: This feels wrong. It's being used as the error message when the create user endpoint returns a conflict status code. This response means there's a conflict with another entity, in this case a user. This means there's an existing user matching the one was attempted to be created.

```suggestion
export const HTTP_CONFLICT_ERROR_MESSAGE: string = 'Cannot register user. There's already a user with the same email address.';
```

File: packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    if (response.statusCode === 200) {
+      setFormStatus(RegisterStatus.backendOK);
+    } else if (response.statusCode === 409) {
+      setBackendError(INVALID_CREDENTIALS_REG_ERROR);
+      setFormStatus(RegisterStatus.backendKO);
+    } else {
+      setBackendError(UNEXPECTED_REG_ERROR);
+      setFormStatus(RegisterStatus.backendKO);
+    }
+  };
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts:126]: use a switch block instead

File: packages/frontend/web-ui/src/auth/pages/Register.spec.tsx [hunk 3]
```diff
@@ -116,17 +120,25 @@ describe(Register.name, () => {
   });
 
   describe('when called, and input value are invalid and an error is displayed', () => {
-    let pErrorName, pErrorEmail, pErrorPassword, pErrorConfirmPassword;
+    let pErrorName: string | null, pErrorEmail: string | null;
+    let pErrorPassword: string | null, pErrorConfirmPassword: string | null;
     const nameFixtureError = 'name Error';
     const emailFixtureError = 'email Error';
     const passwordFixtureError = 'password Error';
     const confirmPasswordFixtureError = 'confirmPassword Error';
```
[the remark this answers is not in the document]
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/pages/Register.spec.tsx:128]: Done! I've created a structure with type FormValidationResult.

File: packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts:126]: Done!

File: packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+import { UseRegisterFormParams } from '../models/UseRegisterFormResult';
+import { RegisterStatus } from '../models/RegisterStatus';
+import { FormValidationResult } from '../models/FormValidationResult';
+import { FormFieldsRegister } from '../models/FormFieldsRegister';
+import { RegisterResponse } from '../../common/http/models/RegisterResponse';
+import { RegisterSerializedResponse } from '../../common/http/models/RegisterSerializedResponse';
+
+export const INVALID_CREDENTIALS_REG_ERROR: string = 'Invalid credentials.';
+export const UNEXPECTED_REG_ERROR: string =
+  'Ups... Something strange happened. Try again?';
```
[the remark this answers is not in the document]
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts:18]: Done!

File: packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/hooks/useRegisterForm.ts:16]: Done!

**Pull request merged**